File: visual-model-ui/src/main/java/org/visual/model/ui/bar/SystemTitleBar.java
package org.visual.model.ui.bar;

import javafx.geometry.Pos;
import javafx.scene.layout.HBox;
import javafx.scene.layout.Priority;

import static org.visual.model.shared.Platform.platform;


public class SystemTitleBar extends HBox {
    {
        CommonTitleBar titleBar = switch (platform) {
            case MAC -> new MacOSTitleBar();
            case LINUX -> new LinuxTitleBar();
            case WINDOWS -> new WindowsTitleBar();
            default -> throw new UnsupportedOperationException("NOT SUPPORT PLATFORM");
        };

        getStylesheets().add("/system.css");
        getStyleClass().add("title-rounded");
        setAlignment(Pos.BASELINE_CENTER);
        setHgrow(titleBar, Priority.ALWAYS);
        getChildren().add(titleBar);
    }
}

File: visual-model-ui/src/main/kotlin/org/visual/model/ui/bar/SystemTitleBar.kt
package org.visual.model.ui.bar

import javafx.geometry.Pos
import javafx.scene.layout.HBox
import javafx.scene.layout.Priority
import lombok.Getter
import lombok.ToString
import org.visual.model.shared.OperationSystem
import org.visual.model.shared.OperationSystem.*
import org.visual.model.shared.detect

@Getter
@ToString
class SystemTitleBar : HBox() {
    private val titleBar = when (detect()) {
        MAC -> MacosTitleBar()
        LINUX -> LinuxTitleBar()
        WINDOWS -> WindowsTitleBar()
        UNKNOWN -> TODO("NOT SUPPORT PLATFORM")
    }

    init {
        stylesheets.add("/system.css")
        styleClass.add("title-rounded")
        alignment = Pos.BASELINE_CENTER
        setHgrow(titleBar, Priority.ALWAYS)
        children.add(titleBar)
    }
}
